### PR TITLE
DSET-2194 migrate CI workflows to GHA

### PR DESCRIPTION
**Pull request by dlobue**
Migrate CI workflows to GHA

DSET-2194

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        uses: docker/login-action@v2
+        with:
+          username: scalyr
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build Docker Image
+        uses: docker/build-push-action@v4
+        with:
+          # set push to true when a tag is created
+          push: ${{ startsWith(github.ref, 'refs/tags/v') }}
```
Comment (jmakar-scalyr) [inline .github/workflows/main.yaml:48]: Do we still want to build if not going push to Docker Hub?

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+      - name: Write RubyGems Credentials to a file
+        if: startsWith(github.ref, 'refs/tags/v')
+        run: ./scripts/set-up-rubygems-credentials.sh
+        env:
+          RUBYGEMS_API_KEY: ${{ secrets.RUBYGEMS_API_KEY }}
+
+      - name: Publish package to RubyGems
+        if: startsWith(github.ref, 'refs/tags/v')
+        run: gem push pkg/fluent-plugin-scalyr-${{ env.VERSION }}.gem
```
Comment (jmakar-scalyr) [inline .github/workflows/main.yaml:63]: Are RubyGems gem versions mutable?  If not, could check for the version and delete before the gem push

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Run lint
+        run: bundle exec rubocop
+
+      - name: Build plugin
+        run: bundle exec rake build
+
+      - name: Run tests
+        run: bundle exec rake test
+
+      - name: Capture version
```
Comment (jmakar-scalyr) [inline .github/workflows/main.yaml:34]: I would add a check to verify the tag matches the version specified in VERSION (and bail if not)

File: test/bad_ca_cert.crt [hunk 4]
```diff
@@ -0,0 +1,20 @@
+-----BEGIN CERTIFICATE-----
```
Comment (jmakar-scalyr) [inline test/bad_ca_cert.crt:1]: What makes this a bad cert? (I was expecting it to be expired)
Used `openssl x509 -in bad_ca_cert.crt -text -noout`

**Review by jmakar-scalyr — APPROVED** [review]
Looks good, had only minor comments

File: test/bad_ca_cert.crt (hunk 4 above)
Comment (dlobue) [reply test/bad_ca_cert.crt:1]: What makes it "bad" is that the server hostname for the cert does not match the hostname that the test is attempting to connect to.

File: .github/workflows/main.yaml (hunk 3 above)
Comment (dlobue) [reply .github/workflows/main.yaml:34]: done.

**Pull request merged**